src/util-inl.h

```diff
@@ -293,6 +293,13 @@ char ToLower(char c) {
   return c >= 'A' && c <= 'Z' ? c + ('a' - 'A') : c;
 }
 
+std::string ToLower(const std::string& in) {
```

Comment:
Random unrelated questions - since we always build with an ICU - isn't this the sort of thing the ICU can do?

Also - is the loop below not a `for(auto elm : in)` due to stylistic preference or something else?

Reply:
> isn't this the sort of thing the ICU can do?

No idea – doing it this way doesn’t cost us much though, though. @srl295 probably has a more helpful answer :)

> Also - is the loop below not a `for(auto elm : in)` due to stylistic preference or something else? 

We’d probably still want to set the characters in the output string directly rather than appending them one at a time, so we’d want to maintain an index for that. In those cases, yes, I prefer to use the explicit index for the input as well, even though we could do it without. :)